spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -169,7 +174,13 @@ public DeltaWriterFactory createBatchWriterFactory(PhysicalWriteInfo info) {
       // broadcast the table metadata as the writer factory will be sent to executors
       Broadcast<Table> tableBroadcast =
           sparkContext.broadcast(SerializableTableWithSize.copyOf(table));
-      return new PositionDeltaWriteFactory(tableBroadcast, command, context, writeProperties);
+      Broadcast<Map<String, DeleteFileSet>> rewritableDeletes = null;
+      if (context.deleteGranularity() == DeleteGranularity.FILE && scan != null) {
+        rewritableDeletes = sparkContext.broadcast(scan.rewritableDeletes());
```

Comment:
We should avoid the broadcast if the set of rewritable deletes is empty/null. I'd also move this into a helper method and modify the comment/invocation. Somel

```
@Override
public DeltaWriterFactory createBatchWriterFactory(PhysicalWriteInfo info) {
  // broadcast large objects as the writer factory will be sent to executors
  return new PositionDeltaWriteFactory(
      sparkContext.broadcast(SerializableTableWithSize.copyOf(table)),
      broadcastRewritableDeletes(),
      ...
}

private Broadcast<Map<String, DeleteFileSet>> broadcastRewritableDeletes() {
  if (context.deleteGranularity() == DeleteGranularity.FILE && scan != null) {
    Map<String, DeleteFileSet> rewritableDeletes = scan.rewritableDeletes();
    if (rewritableDeletes != null && !rewritableDeletes.isEmpty()) {
      return sparkContext.broadcast(rewritableDeletes);
    }
  }

  return null;
}
```

Reply:
Sorry I think I meant to address this from your earlier comment and mistakenly resolved it!

Reply:
Will update